Gems/Compression/Code/Include/Compression/DecompressionInterfaceAPI.h

```diff
@@ -70,28 +70,45 @@ namespace Compression
         //! Invokes the supplied visitor for each register decompression interface that is non-nullptr
         virtual void VisitDecompressionInterfaces(const VisitDecompressionInterfaceCallback&) const = 0;
 
-        //! Registers a decompression interface with the decompression factory.
+
+        //! Registers decompression interface and takes ownership of it if registration is successful
+        //! @param algorithmId Unique id to associate with decompression interface
+        //! @param decompressionInterface decompression interface to register
+        //! @return Success outcome if the decompression interface was successfully registered
+        //! Otherwise, a failure outcome with the decompression interface is forward back to the caller
+        virtual AZ::Outcome<void, AZStd::unique_ptr<IDecompressionInterface>> RegisterDecompressionInterface(
+            CompressionAlgorithmId algorithmId, AZStd::unique_ptr<IDecompressionInterface>) = 0;
+
+
+        //! Registers decompression interface, but does not take ownership of it
         //! If a decompression interface with a CompressionAlgorithmId is registered that
         //! matches the input decompression interface, then registration does not occur
-        //! and the unique_ptr refererence is unmodified and can be re-used by the caller
         //!
-        //! @param decompressionInterface unique pointer to decompression interface to register
-        //! @return true if the IDecompressionInterface was successfully registered, otherwise false
-        virtual bool RegisterDecompressionInterface(AZStd::unique_ptr<IDecompressionInterface>&&) = 0;
+        //! Registers decompression interface, but does not take ownership of it
+        //! @param algorithmId Unique id to associate with decompression interface
+        //! @param decompressionInterface decompression interface to register
+        //! @return true if the ICompressionInterface was successfully registered
+        virtual bool RegisterDecompressionInterface(CompressionAlgorithmId algorithmId, IDecompressionInterface&) = 0;
```

Comment:
Should match doxygen entry for ```decompressionInterface```
```suggestion
        virtual bool RegisterDecompressionInterface(CompressionAlgorithmId algorithmId, IDecompressionInterface& decompressionInterface) = 0;
```